server.go

```diff
@@ -83,6 +83,28 @@ func ServerPingInterval(interval time.Duration) ServerOption {
 	}
 }
 
+// WithClient sets the server to use an existing
+// ExtensionManagerClient instead of creating a new one. This is
+// useful as it causes the ExtensionManagerServer to share mutex locks
+// with an existing client.
+func WithClient(client ExtensionManager) ServerOption {
+	return func(s *ExtensionManagerServer) {
+		s.serverClient = client
+	}
+}
+
+type ExtensionManager interface {
```

Comment:
I'm not sure why this interface is exported. It seems like it should be an internal interface, since it's really declaring a consumer's expectation for what the server expects a client to look like. But I didn't want to change it.

Reply:
I can't remember any good reason for this.